packages/base/src/shared/components/loading.tsx

```diff
@@ -0,0 +1,27 @@
+import { faSpinner } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import React from 'react';
+
+type FontAwesomeIconSize = '2xs' | 'xs' | 'sm' | 'lg' | 'xl' | '2xl' | '1x' | '2x' | '3x' | '4x' | '5x' | '6x' | '7x' | '8x' | '9x' | '10x';
```

Comment:
I hoped this would be importable, but `react-fontawesome` is not written in TypeScript. :(